lndclient/router_client.go

```diff
@@ -0,0 +1,161 @@
+package lndclient
+
+import (
+	"context"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"google.golang.org/grpc/codes"
+
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// RouterClient exposes payment functionality.
+type RouterClient interface {
+	SendPayment(ctx context.Context, invoice string,
+		maxFee btcutil.Amount, maxCltv *int32, outgoingChannel *uint64,
+		timeout time.Duration) (chan PaymentStatus, chan error, error)
+
+	TrackPayment(ctx context.Context, hash lntypes.Hash) (
+		chan PaymentStatus, chan error, error)
+}
+
+// PaymentStatus describe the state of a payment.
+type PaymentStatus struct {
+	State    routerrpc.PaymentState
+	Preimage lntypes.Preimage
+	Fee      lnwire.MilliSatoshi
+	Route    []uint64
```

Comment:
Perhaps we should just use `router.Route` here in order to give the caller all the information available from the response of the call. Although, perhaps we only need the last few channel IDs in our particular use case.

Reply:
Converted to `route.Route`